### PR TITLE
Network xr hands

### PR DESCRIPTION
**Pull request by hamzzam**
## Summary

Networking of XR hands

## Checklist
- [x] CI/CD checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Typescript passing via `npm run check-errors`
  - [x] Unit & Integration tests passing via `npm run test`
  - [ ] Docker build process passing via `npm run build-client`
- [x] If this PR is still a WIP, convert to a draft 
- [x] When this PR is ready, mark it as "Ready for review"
- [x] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## QA Steps
Open 2 instances of a test location.
Set avatar controller type to XR hands from the menu.
Launch VR mode. 
XR Hands movements should be synced across all sessions.

## Reviewers
@HexaField

### CODE REVIEW

File: packages/engine/src/avatar/AvatarSystem.ts [hunk 1]
```diff
@@ -75,6 +77,16 @@ function avatarActionReceptor(action) {
...
 
           addComponent(entity, XRInputSourceComponent, inputData as any)
+
+          dispatchAction(
+            Engine.instance.store,
+            AvatarInputSettingsAction.setControlType(a.avatarInputControllerType as any)
+          )
+          // This is required because state is updated in the next frame and some functionality that requires updated controller type might run in the current frame.
+          // This component can be used at such places.
+          addComponent(entity, AvatarInputControllerTypeUpdatePendingComponent, {
```
Comment (HexaField) [inline packages/engine/src/avatar/AvatarSystem.ts:87]: let's try and refactor around not needing this component, we should rely more heavily on actions than adding and removing components

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/avatar/AvatarSystem.ts (hunk 1 above)
Comment (speigg) [reply packages/engine/src/avatar/AvatarSystem.ts:87]: "some functionality that requires updated controller type might run in the current frame" which functionality?
"This component can be used at such places." What places?

Comment (hamzzam) [reply packages/engine/src/avatar/AvatarSystem.ts]: > "some functionality that requires updated controller type might run in the current frame" which functionality? "This component can be used at such places." What places?

Hey sorry for the confusion I have updated the comment now to be more understandable.

**Comment by dmitrizagidulin** [comment]
PR Moved to https://github.com/XRFoundation/XREngine/pull/6111

**Pull request closed without merging**